internal/core/funcs.go

```diff
@@ -27,3 +28,33 @@ func GetRouteFromUrl(url string) (string, bool) {
 
 	return "", false
 }
+
+func ParseRoute(route string, repoOnly bool) (string, string, string, error) {
+	elements := strings.FieldsFunc(route, func(char rune) bool { return char == '/' })
+	validElementPattern := regexp.MustCompile(`^[\w\.-]+$`)
```

Comment:
I've verified by looking at https://github.com/google/re2/wiki/Syntax that `\w` is equivalent to `[A-Za-z0-9_]`, i.e. it includes the underscore. Good.